tests/render/expectations/test_render_expect_query_results_to_match_source.py

```diff
@@ -0,0 +1,145 @@
+                type="expect_query_results_to_match_source",
+                kwargs={
+                    "target_query": "SELECT * FROM {batch}",
+                    "source_data_source_name": "My Data Source",
+                    "source_query": "SELECT * FROM a_table_in_source_data_source",
+                },
+            ),
+            [
+                {
+                    "name": "atomic.prescriptive.summary",
+                    "value": {
+                        "code_block": {
+                            "code_template_str": "$target_query",
+                            "language": CodeBlockLanguage.SQL,
+                        },
+                        "params": {
+                            "target_query": {
+                                "schema": {"type": RendererValueType.STRING},
+                                "value": "SELECT * FROM {batch}",
+                            }
+                        },
+                        "schema": {"type": "com.superconductive.rendered.string"},
+                    },
+                    "value_type": "StringValueType",
+                },
+                {
+                    "name": "atomic.prescriptive.summary",
+                    "value": {
+                        "code_block": {
+                            "code_template_str": "$source_query",
+                            "language": CodeBlockLanguage.SQL,
+                        },
+                        "params": {
+                            "source_data_source_name": {
+                                "schema": {"type": RendererValueType.STRING},
+                                "value": "My Data Source",
+                            },
+                            "source_query": {
+                                "schema": {"type": RendererValueType.STRING},
+                                "value": "SELECT * FROM a_table_in_source_data_source",
+                            },
+                        },
+                        "schema": {"type": "com.superconductive.rendered.string"},
+                        "template": "Compare with Data Source $source_data_source_name",
+                    },
+                    "value_type": "StringValueType",
+                },
+            ],
+            id="no_description",
+        ),
+    ],
+)
+@pytest.mark.unit
+def test_expectation_configuration_rendered_atomic_content(
+    expectation_configuration: ExpectationConfiguration,
+    expected_expectation_configuration_rendered_atomic_content: dict,
+):
+    inline_renderer: InlineRenderer = InlineRenderer(render_object=expectation_configuration)
+
+    expectation_configuration_rendered_atomic_content: list[RenderedAtomicContent] = (
+        inline_renderer.get_rendered_content()
+    )
+
+    assert len(expectation_configuration_rendered_atomic_content) == 2
+
+    actual_expectation_configuration_rendered_atomic_content: list[dict] = [
+        rendered_atomic_content.to_json_dict()
```

Comment:
Not blocking, But I'd strongly suggest that we a) not set up big dictionaries in the test setup, and the b) not call to_json_dict here, but instead compare the RenderedAtomicContent (or whichever class it is). Similar suggestion with the expectations; we should prefer using Expectation over ExpectationConfiguration wherever we can. I forget if we marked ExpectationConfiguration as deprecated, but that was the idea with v1.